### ISSUE TITLE
pystac 1.0 compatability

### ISSUE THREAD

**Issue by TomAugspurger** [intake/intake-stac#93]
PySTAC is preparing for 1.0, which includes some breaking changes. Running locally, I see some failures.

```
========================================================================================================= ERRORS =========================================================================================================
____________________________________________________________________________________ ERROR at setup of test_cat_from_item_collection _____________________________________________________________________________________
E   pystac.errors.STACTypeError: JSON does not represent a STAC object.
________________________________________________________________________________________ ERROR at setup of test_cat_to_geopandas _________________________________________________________________________________________
E   pystac.errors.STACTypeError: JSON does not represent a STAC object.
_____________________________________________________________________________ ERROR at setup of test_cat_to_geopandas_crs[IGNF:ETRS89UTM28] ______________________________________________________________________________
E   pystac.errors.STACTypeError: JSON does not represent a STAC object.
________________________________________________________________________________ ERROR at setup of test_cat_to_geopandas_crs[epsg:26909] _________________________________________________________________________________
E   pystac.errors.STACTypeError: JSON does not represent a STAC object.
____________________________________________________________________________________ ERROR at setup of test_cat_to_missing_geopandas _____________________________________________________________________________________
E   pystac.errors.STACTypeError: JSON does not represent a STAC object.
_____________________________________________________________________________________ ERROR at setup of test_load_satsearch_results ______________________________________________________________________________________
E   pystac.errors.STACTypeError: JSON does not represent a STAC object.
======================================================================================================== FAILURES ========================================================================================================
/home/taugspurger/src/stac-extensions/intake-stac/intake_stac/catalog.py:302: ValueError: STAC Item must implement "eo" extension to use this method
/home/taugspurger/src/stac-extensions/intake-stac/intake_stac/catalog.py:302: ValueError: STAC Item must implement "eo" extension to use this method
/home/taugspurger/src/stac-extensions/intake-stac/intake_stac/catalog.py:302: ValueError: STAC Item must implement "eo" extension to use this method
/home/taugspurger/src/stac-extensions/intake-stac/intake_stac/tests/test_catalog.py:167: AssertionError: Regex pattern 'ANG not found in list of eo:bands in collection' does not match 'STAC Item must implement "eo" extension to use this method'.
/home/taugspurger/src/stac-extensions/intake-stac/intake_stac/tests/test_catalog.py:174: AssertionError: Regex pattern "'B8', 'B9', 'blue', 'cirrus'" does not match 'STAC Item must implement "eo" extension to use this method'.
/home/taugspurger/src/stac-extensions/intake-stac/intake_stac/catalog.py:302: ValueError: STAC Item must implement "eo" extension to use this method
================================================================================================ short test summary info =================================================================================================
FAILED intake_stac/tests/test_catalog.py::test_cat_item_stacking - ValueError: STAC Item must implement "eo" extension to use this method
FAILED intake_stac/tests/test_catalog.py::test_cat_item_stacking_using_common_name - ValueError: STAC Item must implement "eo" extension to use this method
FAILED intake_stac/tests/test_catalog.py::test_cat_item_stacking_path_as_pattern - ValueError: STAC Item must implement "eo" extension to use this method
FAILED intake_stac/tests/test_catalog.py::test_cat_item_stacking_dims_of_different_type_raises_error - AssertionError: Regex pattern 'ANG not found in list of eo:bands in collection' does not match 'STAC Item must i...
FAILED intake_stac/tests/test_catalog.py::test_cat_item_stacking_dims_with_nonexistent_band_raises_error - AssertionError: Regex pattern "'B8', 'B9', 'blue', 'cirrus'" does not match 'STAC Item must implement "eo" e...
FAILED intake_stac/tests/test_catalog.py::test_cat_item_stacking_dims_of_different_size_regrids - ValueError: STAC Item must implement "eo" extension to use this method
ERROR intake_stac/tests/test_catalog.py::test_cat_from_item_collection - pystac.errors.STACTypeError: JSON does not represent a STAC object.
ERROR intake_stac/tests/test_catalog.py::test_cat_to_geopandas - pystac.errors.STACTypeError: JSON does not represent a STAC object.
ERROR intake_stac/tests/test_catalog.py::test_cat_to_geopandas_crs[IGNF:ETRS89UTM28] - pystac.errors.STACTypeError: JSON does not represent a STAC object.
ERROR intake_stac/tests/test_catalog.py::test_cat_to_geopandas_crs[epsg:26909] - pystac.errors.STACTypeError: JSON does not represent a STAC object.
ERROR intake_stac/tests/test_catalog.py::test_cat_to_missing_geopandas - pystac.errors.STACTypeError: JSON does not represent a STAC object.
ERROR intake_stac/tests/test_catalog.py::test_load_satsearch_results - pystac.errors.STACTypeError: JSON does not represent a STAC object.
========================================================================================= 6 failed, 13 passed, 6 errors in 9.74s =========================================================================================

```

I'll look into these over the next couple days.

**Comment by scottyhq** [comment]
yep, i think this project should only work with STAC>1.0 going forward

**Comment by TomAugspurger** [comment]
Closes by #96
